### PR TITLE
Replace cross-fetch with local polyfill and ponyfill for node

### PR DESCRIPTION
**Pull request by mramad2**
## Summary

Next is now poly-filling fetch with SSR support using node-fetch. Following through with similar approach by switching up @gasket/fetch to use native (or poly-filled) fetch in the browser and a pony-fill for SSR. 

## Changelog

Change-log updates in files changed

## Test Plan

As we now switched from passing through cross-fetch to poly-filling window.fetch for browsers and pony-filling node-fetch for node, we've added tests to validate the following:

- Validating exports: Fetch, Request, Response, Headers
- Sending out Get and Post Requests
- Validating abort/cancel works accordingly

### CODE REVIEW

File: packages/gasket-fetch/node.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const fetch = require('node-fetch');
```
Comment (smehta-godaddy) [inline packages/gasket-fetch/node.js:1]: Won't this fail if consumers don't install `devDependencies`?

File: packages/gasket-fetch/browser.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+exports = window.fetch; // To import fetch from @gasket/fetch
+exports.default = window.fetch; // For TypeScript consumers without esModuleInterop.
+exports.fetch = window.fetch; // To import {fetch} from @gasket/fetch
```
Comment (kinetifex) [inline packages/gasket-fetch/browser.js:4]: I don't think we need this. The API should for the browser module should be the same as for node, which does not support `{ fetch }` import and I don't think there's a case for it.

File: packages/gasket-fetch/browser.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+exports.Headers = window.Headers;
+exports.Request = window.Request;
+exports.Response = window.Response;
+
```
Comment (kinetifex) [inline packages/gasket-fetch/browser.js:8]: Let's make `AbortController` available from this package in browser/node.

```suggestion
exports.AbortController = window.AbortController;

```

File: packages/gasket-fetch/package.json [hunk 4]
```diff
@@ -23,7 +25,8 @@
     "url": "https://github.com/godaddy/gasket/issues"
   },
   "homepage": "https://github.com/godaddy/gasket/tree/master/packages/gasket-fetch#readme",
-  "dependencies": {
-    "cross-fetch": "^3.0.4"
+  "dependencies": {},
+  "devDependencies": {
+    "node-fetch": "^2.6.1"
```
Comment (kinetifex) [inline packages/gasket-fetch/package.json:30]: Should be in `dependencies`

**Review by kinetifex — CHANGES_REQUESTED** [review]

File: packages/gasket-fetch/node.js (hunk 1 above)
Comment (mramad2) [reply packages/gasket-fetch/node.js:1]: Yes, I didn't pay attention on the install. 🤦‍♂️

File: packages/gasket-fetch/test/browser.spec.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import 'whatwg-fetch';
+import { describe, it } from 'mocha';
+
+import fetch, { AbortController, Request, Headers, Response } from '../browser';
```
Comment (kinetifex) [inline packages/gasket-fetch/test/browser.spec.js:4]: To be consistent, destructure these from the imported `fetch` like you did for the server.spec.js
```suggestion
import fetch from '../browser';
const { AbortController, Request, Headers, Response } = fetch;
```

**Review by kinetifex — APPROVED** [review]
Small tweak then 👍🏻

**Review by kawikabader — APPROVED** [review]

**Pull request merged**